### PR TITLE
Task: Python script to crawl a cricket website

### PR DESCRIPTION
**Pull request by Raspberry-beans**
This new python file grabs the data from the cricket website.
	All the links of results of matches have been saved in separate list.

Relates: 7

### CODE REVIEW

**Comment by Raspberry-beans** [comment]
@M-Shan kindly update

File: projects/news_feed/PSL_Matches.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#This code grabs the data from the suggested website
+#and then stores the links of results of all matches in a list called All_Matches_Links
```
Comment (M-Shan) [inline projects/news_feed/PSL_Matches.py]: write proper script header.
e.g.

```
"""
    Script for ....

    :file: <filename.py>
    :platform: Windows, Linux
    :synopsis:
       Script to handle ...

    :author:
        - your name and email
"""
```

File: projects/news_feed/PSL_Matches.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# importing required libraries
+import requests
+from bs4 import BeautifulSoup
```
Comment (M-Shan) [inline projects/news_feed/PSL_Matches.py]: is `bs4` and `panda` added in `conf/requirements.txt` file?

File: projects/news_feed/PSL_Matches.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#This code grabs the data from the suggested website
```
Comment (M-Shan) [inline projects/news_feed/PSL_Matches.py]: change file name to `psl_matches.py`

File: projects/news_feed/PSL_Matches.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#and then stores the links of results of all matches in a list called All_Matches_Links
+
+# importing required libraries
+import requests
+from bs4 import BeautifulSoup
+import pandas as pd
+
+# sending my URL request
+url = "https://www.espncricinfo.com/scores/series/8679/season/2020/pakistan-super-league?view=results"
```
Comment (M-Shan) [inline projects/news_feed/PSL_Matches.py]: you can save variables in a separate file

**Comment by M-Shan** [comment]
> @M-Shan kindly update

@faheem-rehman review is done.
if something is un-clear you can contact.

**Comment by Raspberry-beans** [comment]
@M-Shan kindly guide about how to save variables in separate file. Thanks!

**Comment by M-Shan** [comment]
> @M-Shan kindly guide about how to save variables in separate file. Thanks!

you can do it as done in shared 2 scripts.

please change the script extension from `.txt` to `.py` and then try.
github doesn't allow `.py` file format uploading in reviews :(

[main.txt](https://github.com/M-Shan/TechCrunch/files/5360779/main.txt)
[utils.txt](https://github.com/M-Shan/TechCrunch/files/5360780/utils.txt)

**Comment by M-Shan** [comment]
> > @M-Shan kindly guide about how to save variables in separate file. Thanks!
> 
> you can do it as done in shared 2 scripts.
> 
> please change the script extension from `.txt` to `.py` and then try.
> github doesn't allow `.py` file format uploading in reviews :(
> 
> [main.txt](https://github.com/M-Shan/TechCrunch/files/5360779/main.txt)
> [utils.txt](https://github.com/M-Shan/TechCrunch/files/5360780/utils.txt)

@Raspberry-beans any update?

**Comment by Raspberry-beans** [comment]
Working on it 😌

File: projects/news_feed/PSL_Matches.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+"""
```
Comment (M-Shan) [inline projects/news_feed/PSL_Matches.py:1]: rename this file as `psl_matches.py`

File: conf/requirements.txt [hunk 6]
```diff
@@ -3,3 +3,5 @@
 numpy
 opencv-python
 scipy
+Beautifulsoup
+Pnada
```
Comment (M-Shan) [inline conf/requirements.txt]: **Pnada** --> **Panda**

**Review by M-Shan — CHANGES_REQUESTED** [review]
delete `projects/news_feed/__pycache__/utilities.cpython-37.pyc` 
add `__pycache__` in `.gitignore` file

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
+The files with following extension should be ignored.
+
+*.pyc
```
Comment (M-Shan) [inline .gitignore]: instead of this you can add `__pycache__` which will ignore all python compiled files

File: conf/requirements.txt [hunk 8]
```diff
@@ -4,4 +4,4 @@ numpy
 opencv-python
 scipy
 Beautifulsoup
-Pnada
+Panda
```
Comment (M-Shan) [inline conf/requirements.txt]: apologies i mentioned it wrong 

this module name is `pandas`

**Review by M-Shan — CHANGES_REQUESTED** [review]

**Review by M-Shan — APPROVED** [review]
Good job

**Pull request merged**